src/components/EditableToDo.vue

```diff
@@ -151,46 +119,87 @@
 </template>
 
 <script setup lang="ts">
-  import { computed, ref } from 'vue';
-  import { Priority, ToDoItem, Status } from '../models/todoitem-model';
+  import { computed, ref, PropType } from 'vue';
+  import { Priority, ToDoItem } from '../models/todoitem-model';
   import { formatShortDate } from '../utils/date-formatting';
   import { getItemPriority } from '../utils/item-priority';
+  import ModalButton from './ModalButton.vue';
 
-  const props = defineProps<{
-    item?: ToDoItem;
-    index?: number;
-  }>();
+  const props = defineProps({
+    item: Object as PropType<ToDoItem>,
+    index: {
+      type: Number,
+      default: -1,
+    },
+  });
 
   const emit = defineEmits<{
-    (e: 'removeItem', ind: number): void;
+    (e: 'removeItem', ind: Number): void;
     (e: 'addToDo', newToDo: ToDoItem): void;
     (e: 'hideForm'): void;
-    (e: 'updateToDo', ind: number, newToDo: ToDoItem): void;
+    (e: 'updateToDo', ind: Number, newToDo: ToDoItem): void;
   }>();
 
-  const showDropDown = ref(false);
+  const isDropDownShown = ref(false);
 
-  const currentTitle = ref(props.item ? props.item.title : '');
+  const currentOpacity = computed(() => {
+    if (isDropDownShown.value) {
+      return 'opacity-30' as const;
+    }
+    return '' as const;
+  });
 
-  const currentText = ref(props.item ? props.item.text : '');
+  const currentTitle = ref(props.item?.title || '');
 
-  const indexOfThisItem = ref(
-    props.index || props.index === 0 ? props.index : -1
-  );
+  const currentText = ref(props.item?.text || '');
 
-  const currentPriority = ref(props.item ? props.item.priority : Priority.low);
+  const currentPriority = ref(props.item?.priority || Priority.low);
 
   const priorityClass = computed(() => getItemPriority(currentPriority.value));
 
   const date = new Date();
 
   const priorityList = [Priority.low, Priority.medium, Priority.high];
 
-  const showPopUp = ref(false);
+  const isPopUpShown = ref(false);
+
+  const positionOfDropDownButton = computed(() => {
+    if (isPopUpShown.value) {
+      return '' as const;
+    }
+    return 'relative';
+  });
+
+  const dropDownStyle = computed(() => {
+    if (isDropDownShown.value === true) {
+      return 'text-black bg-white border-2 border-black' as const;
+    }
+    return priorityClass.value;
```

Comment:
Don't need this variable if you using just once in this component you can simply return `getItemPriority(currentPriority.value)`